### PR TITLE
Lift to upstream b2251 and update build destinations

### PR DESCRIPTION
**Pull request by philippzagar**
# Lift to upstream b2251 and update build destinations

## :recycle: Current situation & Problem
Out of sync with upstream `b2251`. Only limited to `iphoneos` and `iphoneossimulator` build destinations

## :gear: Release Notes 
- Lift to llama.cpp upstream `b2251`
- Broaden the build destinations to visionOS and macOS 

## :books: Documentation
--

## :white_check_mark: Testing
Tested manually

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: llama.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -431,11 +440,16 @@
 				PRODUCT_BUNDLE_IDENTIFIER = edu.stanford.llama;
 				PRODUCT_NAME = "$(TARGET_NAME:c99extidentifier)";
 				SKIP_INSTALL = YES;
+				SUPPORTED_PLATFORMS = "iphoneos iphonesimulator macosx xros xrsimulator";
```
Comment (PSchmiedmayer) [inline llama.xcodeproj/project.pbxproj:443]: Nice 🎉

**Review by PSchmiedmayer — APPROVED** [review]
Thank you @philippzagar 🚀

**Comment by philippzagar** [comment]
@PSchmiedmayer Please force merge the PR as some commits from upstream are unsigned :)

**Comment by PSchmiedmayer** [comment]
Done 🚀

**Pull request merged**